### PR TITLE
xtask cheatsheets: add unit tests and handle slide deck title corner case

### PR DESCRIPTION
**Pull request by miguelraz**
This PR adds the functionality so that the cpp cheatsheet skeleton in #250 can be properly produced, tested, and maintained.

TL;DR:

Before, we would write out the slides like this for a `cpp-cheatsheet.md`:

```text
# Rust Fundamentals
## Overview
## Basic Types
...
```

Whereas it needed to be like this

```text
# Cpp Cheatsheet
# Rust Fundamentals
## Overview
## Basic Types
```

so that the slide form was properly displayed.

The rest of the PR is the unit tests to justify said change and refactoring of some logic into its own function to facilitate said tests.

### CODE REVIEW

File: xtask/src/tasks.rs [hunk 1]
```diff
@@ -227,13 +223,25 @@ pub fn test_cheatsheet(lang: &str) -> Result<(), eyre::Report> {
     let file_name = format!("./training-slides/src/{lang}-cheatsheet.md");
     let cheatsheet_text = read_to_string(file_name)
         .with_context(|| eyre::eyre!("{lang}-cheatsheet.md not found."))?;
+    
+    cheatsheet_tester(lang, slide_sections, &cheatsheet_text)
+}
+
+fn cheatsheet_tester(lang: &str, slide_sections: Vec<SlidesSection>, cheatsheet_text: &str) -> Result<(), eyre::Error> {
+    // Safety: We know `lang` is part of our blessed lang-names, so it must be at least of size 1.
```
Comment (jonathanpallant) [inline xtask/src/tasks.rs:231]: Not technically a safety issue because out-of-bounds array access will panic, but this is still good to note.

File: xtask/src/tasks.rs [hunk 2]
```diff
@@ -227,13 +223,25 @@ pub fn test_cheatsheet(lang: &str) -> Result<(), eyre::Report> {
     let file_name = format!("./training-slides/src/{lang}-cheatsheet.md");
     let cheatsheet_text = read_to_string(file_name)
         .with_context(|| eyre::eyre!("{lang}-cheatsheet.md not found."))?;
+    
+    cheatsheet_tester(lang, slide_sections, &cheatsheet_text)
+}
+
+fn cheatsheet_tester(lang: &str, slide_sections: Vec<SlidesSection>, cheatsheet_text: &str) -> Result<(), eyre::Error> {
```
[the remark this answers is not in the document]
Comment (jonathanpallant) [reply xtask/src/tasks.rs:230]: Also, what's the difference between `fn test_cheatsheet` and `fn cheatsheet_tester`?

**Review by jonathanpallant — CHANGES_REQUESTED** [review]
I could use a little more clarity in what these changes do.

File: xtask/src/tasks.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (miguelraz) [reply xtask/src/tasks.rs:230]: Good point. Added  `#![deny(missing_docs)]` to `main.rs`.

I wanted to write tests for `test_cheatsheet`, but that would involve writing to temp files and tearing them down in each unit test, and that seems like overkill.

I decided instead to refactor out the logic to its own function instead to have more modular unit tests and call this function `setup_cheatsheet_tester`.

File: xtask/src/tasks.rs (hunk 1 above)
Comment (miguelraz) [reply xtask/src/tasks.rs]: Neat - I remember you had written a notice like that in some other code and it seemed like good practice, so I stole it.
